pages/model/[uuid].tsx

```diff
@@ -108,12 +111,47 @@ function Model() {
     setCopyModelCardSnackbarOpen(false)
   }
 
+  useEffect(() => {
```

Comment:
Could be shortened to
```
  useEffect(() => {
    if (currentUser && version) {
      if (currentUser.id === version.metadata.contacts.manager) {
        updateLastViewed('manager')
      }
      if (currentUser.id === version.metadata.contacts.reviewer) {
        updateLastViewed('reviewer')
      }
    }
  }, [managerLastViewed, reviewerLastViewed, currentUser])
```

It looks like you're missing `version` and `updateLastViewed` as a dependencies though. I'd recommend wrapping `updateLastViewed` with `useCallback` and moving it above this `useEffect` to avoid triggering this effect on every render and to avoid the no-use-before-define warnings.